### PR TITLE
enabled Symfony 3.0 support

### PR DESCRIPTION
**Pull request by lsmith77**
also tweaked some aspects of the travis configuration and the composer requires

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,9 +12,10 @@
         }
     ],
     "require": {
-        "php": ">=5.3.2",
+        "php": ">=5.4.0",
         "jms/serializer": "^1.0.0",
-        "symfony/framework-bundle": "~2.1"
+        "phpoption/phpoption": "^1.0.0",
```
Comment (lsmith77) [inline composer.json]: this probably should have been in the ``jms/serializer``

File: composer.json [hunk 2]
```diff
@@ -12,9 +12,10 @@
         }
     ],
     "require": {
-        "php": ">=5.3.2",
+        "php": ">=5.4.0",
         "jms/serializer": "^1.0.0",
-        "symfony/framework-bundle": "~2.1"
+        "phpoption/phpoption": "^1.1.0",
```
Comment (lsmith77) [inline composer.json]: this probably should also be set ``jms/serializer``

**Comment by lsmith77** [comment]
https://travis-ci.org/schmittjoh/JMSSerializerBundle/jobs/89742752

```
Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: You cannot set service "service_container".
/home/travis/build/schmittjoh/JMSSerializerBundle/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:169
/home/travis/build/schmittjoh/JMSSerializerBundle/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:364
/home/travis/build/schmittjoh/JMSSerializerBundle/Tests/DependencyInjection/JMSSerializerExtensionTest.php:168
/home/travis/build/schmittjoh/JMSSerializerBundle/Tests/DependencyInjection/JMSSerializerExtensionTest.php:63
```

**Comment by lsmith77** [comment]
@schmittjoh would be nice to merge this and tag 1.1 :)

**Comment by drgomesp** [comment]
@schmittjoh would love that to happen asap :+1:

**Comment by weaverryan** [comment]
:+1: it's important to have this merged + a tag so that we can update other dependent bundles (e.g. FOSRestBundle) for 3.0 compat. It's a big domino effect.

**Pull request merged**